File: ProductManagementSystem2/src/main/java/com/productManagement/dto/OfferDetailsDTO.java
package com.productManagement.dto;

import java.time.LocalDate;

import com.productManagement.enums.OfferStatus;
import com.productManagement.enums.OfferValue;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class OfferDetailsDTO {

    private Long id;
    private String code;
    private String description;
    private OfferValue value;
    private OfferStatus status;
    private LocalDate activationDate;
    private LocalDate expiryDate;

}

File: FrontendOfSpringProject/src/app/models/offer-details-dto.ts

export interface OfferDetailsDto {
    id: number;
    code: string;
    description: string;
    value: string; // Adjust based on your enum types if necessary
    status: string; // Adjust based on your enum types if necessary
    activationDate: string; // Use Date if needed
    expiryDate: string;
  }
  